src/make_data/banner_intervention/banner_journey_results_link.sql

```diff
@@ -0,0 +1,111 @@
+/*
+This query extracts the total number of journeys that engage with the results page (i.e.
+selects at least 1 link on the results page) and the two SaB pagePaths that trigger
+the banner intervention.
+
+- First, the query creates a table of all disinct sessions that select the banner
+  and selects at least 1 link on the results page of the checker (sessions_results_page).
+- Next, the query only keeps SaB PAGE hits and the EVENT action 'interventionClicked'
+  from those sessions as defined above.
+- Then, in hit order, the query calculates the previous pagePath of the EVENT action
+  'interventionClicked' by counting 2 pagePaths back. NB: we look at 2 pagePaths
+  back as only 1 pagePath is the same pagePath (the PAGE hit) as the EVENT hit
+  'interactionClicked'
+- Count the number of journeys for each pagePath and previousPagePath combination
+  for 'interventionClicked' hits only (i.e. provide the pagePath combination that
+  leads to the banner being selected).
+
+No requirements to run this script. Uses `govuk-bigquery-analytics.87773428.ga_sessions_*`
+output table.
+*/
+
+
+DECLARE start_date STRING DEFAULT "20210803";
+DECLARE end_date STRING DEFAULT "20210816";
+
+CREATE OR REPLACE TABLE `govuk-bigquery-analytics.banner_intervention.banner_journey_results_link` AS
+
+-- All distinct sessions that select the checker via the banner (eventAction =
+-- interventionClicked)
+
+WITH sessions_select_banner AS (
+  SELECT DISTINCT
+    CONCAT(fullVisitorId, "-", visitId) AS sessionId,
+    hits.eventInfo.eventAction
+  FROM `govuk-bigquery-analytics.87773428.ga_sessions_*`
+  CROSS JOIN UNNEST(hits) AS hits
+  WHERE _TABLE_SUFFIX BETWEEN start_date AND end_date
+    AND hits.eventInfo.eventAction  = 'interventionClicked'
+ ),
+
+-- All distinct sessions that select the checker via the banner that also visit
+-- at least 1 result link on the results page of the checker.
+
+sessions_results_page AS (
+  SELECT DISTINCT
+     CONCAT(fullVisitorId, "-", visitId) AS sessionId
+  FROM `govuk-bigquery-analytics.87773428.ga_sessions_*`
+  CROSS JOIN UNNEST(hits) AS hits
+  WHERE _TABLE_SUFFIX BETWEEN start_date AND end_date
+    AND hits.eventInfo.eventCategory = 'SmartAnswerClicked'
+    AND CONCAT(fullVisitorId, "-", visitId) IN (SELECT sessionId FROM sessions_select_banner)
+ ),
+
+-- During those sessions which get to the end of the checker (i.e. the results page),
+-- filter by SaB pages
+
+sessions_journey AS (
+  SELECT
+    TIMESTAMP_MILLIS(CAST(hits.time + (visitStartTime * 1000) AS INT64)) AS datetime,
```

Comment:
I think `visitStartTime` may be constant for the whole visit/session, in which case perhaps in future we could order just by `hits.time`

Reply:
Good to know!